src/products/views/ProductVariant.tsx

```diff
@@ -130,16 +130,22 @@ export const ProductVariant: React.FC<ProductUpdateProps> = ({
     variant: ProductVariantDetails_productVariant
   ) => {
     if (
-      data.channelListing.some(
-        (channel, index) =>
-          channel.price !== variant.channelListing[index]?.price.amount
-      )
+      data.channelListing.some(channel => {
+        const variantChannel = variant.channelListing.find(
+          variantChannel => variantChannel.channel.id === channel.id
+        );
+        return (
+          channel.price !== variantChannel?.price?.amount ||
+          channel.costPrice !== variantChannel?.costPrice?.amount
+        );
+      })
```

Comment:
It's a little bit hard to understand at the first sight what this condition mean. IMO it would be better to extract it as variable e.g. `isChannelPriceChange` and put it in if condition.